Content.Shared/Bed/Cryostorage/SharedCryostorageSystem.cs

```diff
@@ -148,7 +158,11 @@ private void OnShutdownContained(Entity<CryostorageContainedComponent> ent, ref
     {
         var comp = ent.Comp;
 
-        CompOrNull<CryostorageComponent>(comp.Cryostorage)?.StoredPlayers.Remove(ent);
+        // Try to get the lost and found and remove the player from it
+        var query = EntityQueryEnumerator<LostAndFoundComponent>();
+        query.MoveNext(out var storage, out var lostAndFoundComponent);
+        CompOrNull<LostAndFoundComponent>(storage)?.StoredPlayers.Remove(ent);
```

Comment:
I don't think this would work with multiple storages either, find which storage the entity is in and remove itself.